app/models/katello/content_view.rb

```diff
@@ -579,14 +579,16 @@ def check_docker_repository_names!(environments)
 
     def check_composite_action_allowed!(env)
       if composite? && Setting['restrict_composite_view']
-        # verify that the composite's component view versions exist in the target environment.
+        env_ids = env.try(:pluck, 'id') || []
+        env_ids << env.id unless env_ids.size > 0
         components.each do |component|
-          unless component.environments.include?(env)
+          component_environment_ids = component.environments.try(:pluck, 'id')
```

Comment:
same question as above

Reply:
@johnpmitsch : I think I was just being over cautious.. I agree it is safe to assume presence of env  and component.environments in the flow.